madgraph/interface/common_run_interface.py

```diff
@@ -7915,7 +7932,63 @@ def new_fct(obj, *args, **opts):
                     set_run_name(obj)(next_name)
                     # run for the first time
                     original_fct(obj, *args, **opts)
-                    param_card_iterator.store_entry(next_name, store_for_scan(obj)(), param_card_path=card_path)
+                    # check whether mg5/MadEvent or mg5amc/MC@NLO type event
+                    if isinstance(input_path, str):
+                        run_path = run_card_input
+                    else:
+                        run_path = run_card_input(obj)
+                    run_card_iterator = run_card_iteratorclass(run_path)
+                    ismg5amc= not run_card_iterator.run_card.LO
```

Comment:
```suggestion
                    # check whether mg5/MadEvent or mg5amc/MC@NLO type event
                    ismg5amc = not self.LO
```
Would this not work?

Reply:
"ismg5amc = not self.LO" does not work because the overarching structure is the function `scanparamcardhandling' and not a class, so "self" does not exist (if I understand scope/logic correctly)

Even if I tried, I get the following NameError:
print(f"{self.LO=}")
             ^^^^
NameError: name 'self' is not defined

*maybe* it is possible to use `MADEVENT=True    / MADEVENT=False' which are defined at the module level in the import lines, but I have not tried this

Reply:
either way, fixed! see the following commit for details https://github.com/mg5amcnlo/mg5amcnlo/tree/d44d651353aa1daf712d4d77feed91d7ad69692d

Reply:
additional commit for improved exception handling (per above)
https://github.com/mg5amcnlo/mg5amcnlo/tree/fcba1b8d72cdd75eb7b7f9487eeb8ddf13e56e2a